ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -169,8 +170,25 @@ private[kvutils] class TransactionCommitter(
                     RejectionReason.InvalidLedgerTime(reason))),
               _ => StepContinue(transactionEntry)
             )
-        }
-        .getOrElse(StepContinue(transactionEntry))
+        case _ => // Pre-execution: propagate the time bounds and defer the checks to post-execution
+          val maybeDeduplicateUntil =
+            PartialFunction.condOpt(transactionEntry.submitterInfo.hasDeduplicateUntil) {
```

Comment:
You don't want to take the deduplication time from the submitted transaction, but from what is stored on the ledger in the deduplication key for the command ID of the submission.

Reply:
Thanks, addressed.